### PR TITLE
Feature/process data

### PR DESCRIPTION
**Pull request by Metaverse-Nabeel**
### The following changes have been made to this PR:

**Set up associations between models**

- Established a foreign key association between the users and posts tables in the models.
- Ensured that the author_id column in the posts table references the id column in the users table.

**Used models to insert data**

- Utilize the Rails console (rails c) to interact with the project.
- Created Users
- Generate more than 4 posts written by one of the created users
- Implemented a method that returns the three most recent posts for a given user in User.rb file.
- Developed a method that updates the post counter for a user in the User.rb inside Post.rb file.
- Designed a method that returns the five most recent comments for a given post inside the Post.rb file.
- Create a method that updates the comments counter for a post in the Comment.rb file.
- Constructed a method that updates the likes counter for a post in the Like.rb file .

**Verify Functionality**

- Access the Rails console (rails c) to validate the functionality of the implemented methods.

- Corrected Linter Errors
- Updated Readme File

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class Like < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
+  belongs_to :post, foreign_key: 'post_id'
+
+  def increment_like_counter
+    puts 'Incrementing like counter'
+    user.increment!(:likes_counter)
+  end
```
Comment (tugsem) [inline app/models/like.rb]: - [x] Please increment the likes_counter for a post, not the user as the likes should belong to a post, not a user.

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,2 +1,9 @@
 class Comment < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
+  belongs_to :post, foreign_key: 'post_id'
+
+  def increment_comment_counter
+    puts 'Incrementing comment counter'
+    user.increment!(:comments_counter)
```
Comment (tugsem) [inline app/models/comment.rb]: - [x] Please increment the comments_counter for the post, not the user as the comments should belong to a post.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @Metaverse-Nabeel 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️ Associations between models are set up correctly
✔️required methods are written
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @Metaverse-Nabeel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Good job on implementing the changes ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**